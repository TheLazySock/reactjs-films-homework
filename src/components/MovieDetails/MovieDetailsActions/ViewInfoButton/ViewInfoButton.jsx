import React from 'react';
import PropTypes from 'prop-types';
import style from './ViewInfoButton.scss';

const ViewInfoButton = (props) => {
  const { onClick } = props;

  return (
    <button type="button" className={style.viewInfoBtn} onClick={onClick}>View Info</button>
  );
};

ViewInfoButton.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default ViewInfoButton;
