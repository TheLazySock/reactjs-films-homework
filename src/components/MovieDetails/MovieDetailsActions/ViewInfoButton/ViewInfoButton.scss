.viewInfoBtn {
  padding: 7px 15px;
  border: 1px solid white;
  border-radius: 5px;
  color: white;
  cursor: pointer;
  font-size: 14px;
  background-color: #000;

  &:hover {
    background-color: #313131;
  }

  &:active {
    background-color: #000;
  }
}
