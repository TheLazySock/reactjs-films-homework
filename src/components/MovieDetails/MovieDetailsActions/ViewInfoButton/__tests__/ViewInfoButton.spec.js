import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import ViewInfoButton from '../index';

describe('ViewInfoButton', () => {
  it('renders properly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<ViewInfoButton onClick={jest.fn()} />);
    const output = renderer.getRenderOutput();
    expect(output.type).toBe('button');
    expect(output).toMatchSnapshot();
  });
});
