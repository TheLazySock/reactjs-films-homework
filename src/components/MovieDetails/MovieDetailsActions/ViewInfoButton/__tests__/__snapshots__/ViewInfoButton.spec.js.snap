// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewInfoButton renders properly 1`] = `
<button
  className="viewInfoBtn"
  onClick={[MockFunction]}
  type="button"
>
  View Info
</button>
`;
